cargo.sh

```diff
@@ -1,135 +1,12 @@
-  #
-  # In CI, make sure to use the default stable toolchain. If we're testing on
-  # our MSRV, then we also have our MSRV toolchain installed. As of this
-  # writing, our MSRV is low enough that the correspoding Rust toolchain's Cargo
-  # doesn't know about the `rust-version` field, and so if we were to use Cargo
-  # with that toolchain, `pkg-meta` would return `null` when asked to retrieve
-  # the `rust-version` field. This also requires `RUSTFLAGS=''` to override any
-  # unstable `RUSTFLAGS` set by the caller.
-  RUSTFLAGS='' CARGO_TARGET_DIR=target/cargo-sh cargo +stable metadata --format-version 1 | jq -r ".packages[] | select(.name == \"zerocopy\").$1"
-}
-
-function lookup-version {
-  VERSION="$1"
-  case "$VERSION" in
-    msrv)
-      pkg-meta rust_version
-      ;;
-    stable)
-      pkg-meta 'metadata.ci."pinned-stable"'
-      ;;
-    nightly)
-      pkg-meta 'metadata.ci."pinned-nightly"'
-      ;;
-    *)
-      TOOLCHAIN=$(pkg-meta "metadata.\"build-rs\".\"${VERSION}\"")
-      if [ "$TOOLCHAIN" != "null" ]; then
-        echo "$TOOLCHAIN"
-      else
-        echo "Unrecognized toolchain name: '$VERSION' (options are 'msrv', 'stable', 'nightly', and any value in Cargo.toml's 'metadata.build-rs' table)" >&2
-        return 1
-      fi
-      ;;
-  esac
-}
-
-function get-rustflags {
-  [ "$1" == nightly ] && echo "--cfg __INTERNAL_USE_ONLY_NIGHLTY_FEATURES_IN_TESTS"
-}
-
-function prompt {
-  PROMPT="$1"
-  YES="$2"
-  while true; do
-    read -p "$PROMPT " yn
-    case "$yn" in
-      [Yy]) $YES; return $?; ;;
-      [Nn])       return 1;  ;;
-      *)          break;     ;;
-    esac
-  done
-}
-
-case "$1" in
-  # cargo.sh --version <toolchain-name>
-  --version)
-    [[ $# -eq 2 ]] || print-usage-and-exit
-    lookup-version "$2"
-    ;;
-  # cargo.sh +all [...]
-  +all)
-    echo "[cargo.sh] warning: running the same command for each toolchain (msrv, stable, nightly)" >&2
-    for toolchain in msrv stable nightly; do
-      echo "[cargo.sh] running with toolchain: $toolchain" >&2
-      $0 "+$toolchain" ${@:2}
-    done
-    exit 0
-    ;;
-  # cargo.sh +<toolchain-name> [...]
-  +*)
-    TOOLCHAIN="$(lookup-version ${1:1})"
-
-    cargo "+$TOOLCHAIN" version &>/dev/null && \
-    rustup "+$TOOLCHAIN" component list | grep '^rust-src (installed)$' >/dev/null || {
-      echo "[cargo.sh] missing either toolchain '$TOOLCHAIN' or component 'rust-src'" >&2
-      # If we're running in a GitHub action, then it's better to bail than to
-      # hang waiting for input we're never going to get.
-      [ -z ${GITHUB_RUN_ID+x} ] || exit 1
-      prompt "[cargo.sh] would you like to install toolchain '$TOOLCHAIN' and component 'rust-src' via 'rustup'?" \
-        "rustup toolchain install $TOOLCHAIN -c rust-src"
-    } || exit 1
-
-    RUSTFLAGS="$(get-rustflags ${1:1}) $RUSTFLAGS" cargo "+$TOOLCHAIN" ${@:2}
-    ;;
-  *)
-    print-usage-and-exit
-    ;;
-esac
+# Build `cargos` without any RUSTFLAGS set in the environment
```

Comment:
Can the tool itself now be called `cargo` instead of `cargos`? Or is there a name conflict in doing that?

Reply:
Even if there's not a name conflict, it sounds like naming it `cargo` is asking for trouble. Would something like `cargo-multiplex`/`cargo-multi`/`multicargo`/`mc` be acceptable?

Reply:
Hmm yeah good point. How about something like:
- `cargo-zerocopy`
- `cargo-zerocopy-shim`

Given that this is always called from a script, I'm not too worried about a verbose name.

Reply:
I went with `cargo-zerocopy`